tests/effects/tdiagnostic_messages.nim

```diff
@@ -0,0 +1,32 @@
+discard """
+  errormsg: "\'a\' can have side effects"
+  nimout: '''  tdiagnostic_messages.nim(32, 30) Hint: 'a' calls 'callWithSideEffects' which has side effects
+    tdiagnostic_messages.nim(24, 6) Hint: 'callWithSideEffects' called by a `noSideEffect` routine, but has side effects
+      tdiagnostic_messages.nim(26, 34) Hint: 'callWithSideEffects' calls 'indirectCallViaVarParam' which has side effects
+        tdiagnostic_messages.nim(20, 6) Hint: 'indirectCallViaVarParam' called by a `noSideEffect` routine, but has side effects
```

Comment:
```suggestion
        tdiagnostic_messages.nim(20, 6) Hint: 'indirectCallViaVarParam' called by a `. sideEffect` routine
```
simpler = better.

ditto with other messages

Reply:
```Hint: 'indirectCallViaVarParam' called by a `.sideEffect` routine``` is confusing to me.
What does `.sideEffect` mean?

Reply:
`sideEffect` is a valid pragma:
```nim
proc echo*(x: varargs[typed, `$`]) {.magic: "Echo", tags: [WriteIOEffect], benign, sideEffect.}
```

and `.foo` is a shortcut for a `{.foo.}` pragma that is used in a growing number of places, it's a convention. ask araq :)

Reply:
I wanted to communicate that 'indirectCallViaVarParam'  is trying to be used by a noSideEffect routine. I guess I'm implying a fix of changing 'indirectCallViaVarParam' to be `.noSideEffect` or that the `.noSideEffect` caller needs to be changed to `.sideEffect`.

How would you communicate this to  a new Nim programmer?

Reply:
it should be obvious if you read the whole error message, which has all the context.
The problem with overly verbose error messages is it drowns the essential part (the things that vary from instance to instance) into noise (the things that don't vary). So you might help a tiny bit a confused user the the 1st time they read it, but you'll hurt readability for the next N times you'll see this error msg.